### PR TITLE
feat: tags

### LINKED ISSUE #10: Tag support
A major feature that would be worth considering support for would be arbitrary tags to help categorize posts.

High level scope:
- Determine the syntax to be used within the markdown document i.e. `tags: tag1, tag2, tag3`
- What URL structure for the index of tags along with all the documents tagged something i.e. `/tags` and either `/tags/{tag}` or `/tagged/{tag}`
- Roughly how the data structure would look internally to track tags to their source documents. Will certainly be janky because Bash doesn't make that trivial.

An initial proposal of a hypothetical markdown document:

```markdown
# My title
tags: foo, bar, baz

Here are some words.

Lorem ipsum.
```

I'm partial to `/tags` and `/tags/{tag}` since it's more of a RESTful syntax.

Open questions:
- How to parse tags? 
	- Split on commas, trim whitespace? Do we trim all whitespace? And lowercase?

### PR DESCRIPTION
**Pull request by Pinjasaur**
closes #10

This adds tag support as a means of organizing entries (effectively just posts).

I landed on a `tags: ...` string immediately below the title:

```md
# My Title
tags: one, and-two, and-even-three

And the rest of the normal body goes here.
```

Internally, the mappings get stored as associative arrays. The posts get delimited by `|` characters, so those are now invalid characters in a filename and will cause bic to error out at build-time.

All the tags are available on a `tags.html` view and _also_ use a `tag.html` and `__tag.html` to create `tags/${tag}.html` views to showing the mapping of individual tags to posts.

Notes:

- Drafts and pages cannot be tagged (drafts makes sense, pages could be an enhancement in the future)
- Not extensively tested nor documented

### CODE REVIEW

**Comment by Pinjasaur** [comment]
cc @mfossen

**Review by mfossen — APPROVED** [review]

File: bic [hunk 1]
```diff
@@ -301,9 +382,13 @@ build() {
   [[ -d "${DEST_DIR}" ]] && rm -rf "${DEST_DIR}"
   mkdir -p "${DEST_DIR}" && log "Created build directory: ${DEST_DIR}"
 
+  # Need to declare this slightly more globally otherwise can't share it across functions 🤷
+  declare -A ALL_TAGS
```
Comment (Pinjasaur) [inline bic:401]: unironically caused me to scratch my head longer than I cared to admit

dunno why, but other variables just seem to work out of the box, but something with the `declare` syntax makes that not the case here

File: bic [hunk 2]
```diff
@@ -69,7 +69,26 @@ mk_title() {
 }
 
 mk_body() {
-  pandoc "${PANDOC_ARGS[@]}" <(tail -n +2 "${1}")
+  local prefix tagline
+  prefix="tags:"
+  tagline="$(head -n 2 < "${1}" | tail -n 1)"
+  if [[ "$tagline" == "$prefix"* ]]; then
```
Comment (Pinjasaur) [inline bic]: nit: would like to see this refactored since it's hella duplicated logic

**Comment by Pinjasaur** [comment]
Need to also update the sitemap.xml.

File: docs/index.md [hunk 3]
```diff
@@ -44,29 +45,30 @@ nix run github:Pinjasaur/bic --command bic .
...
-    signify the top-level heading.
+  - _Ordering_ is determined by a number prefix e.g., `999-post.md`
+  for the first post, `998-tacocat.md` for the second, et cetera. I would
+  recommend 3 or 4 digits for the Future Proof&trade;.
+  - This lets the file `mtime` be used for the author's discretion. However,
+  Git [doesn't record `mtime`][mtime], so I would treat it as the "last
+  modified" date.
+  - The title is derived from the _first line_ which _must_ begin with `#` to
+  signify the top-level heading.
+  - Entries can be organized via tags, which much be defined _immediately_
```
Comment (Pinjasaur) [inline docs/index.md]:
```suggestion
  - Entries can be organized via tags, which MUST be defined _immediately_
```

File: docs/index.md [hunk 4]
```diff
@@ -44,29 +45,30 @@ nix run github:Pinjasaur/bic --command bic .
...
-    modified" date.
-    - The title is derived from the _first line_ which _must_ begin with `#` to
-    signify the top-level heading.
+  - _Ordering_ is determined by a number prefix e.g., `999-post.md`
+  for the first post, `998-tacocat.md` for the second, et cetera. I would
+  recommend 3 or 4 digits for the Future Proof&trade;.
+  - This lets the file `mtime` be used for the author's discretion. However,
+  Git [doesn't record `mtime`][mtime], so I would treat it as the "last
+  modified" date.
+  - The title is derived from the _first line_ which _must_ begin with `#` to
```
Comment (Pinjasaur) [inline docs/index.md]:
```suggestion
  - The title is derived from the _first line_ which MUST begin with `#` to
```

**Pull request merged**